### DISCUSSION TITLE
CCIncrementDecrementButtons example

### DISCUSSION THREAD

**Discussion by MickeyGentry** [tttapa/Control-Surface#919] [General]
I have the CCIncrementDecrementButtons example working as expected. When I view the output from MidiView I see the same values output when pressing the buttons. I guess I expected to see higher and lower values for each press. The note on is also working as expected. 

Am I missing something about testing this? Is MidiView not 'seeing' MCU values?

Thanks!
![Screenshot 2023-04-27 at 1 10 09 PM](https://user-images.githubusercontent.com/130603709/234938055-c91aff16-88f2-44d8-927b-53bc685c6362.png)

**Comment by MickeyGentry** [comment]
FYI, I made a sketch from one of your examples for program change using Increment/Decrement, worked great. Not really interested in CC messages right now but thought you might want to know.

**Comment by tttapa** [comment]
`CCIncrementDecrementButtons` send _relative_ control change messages, i.e. they only send the change/delta relative to the previous position.

Different conventions for encoding negative numbers exist, see https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d24/RelativeCCSender_8hpp.html#a793b4d05f50a2c3af3ddeeef50e7a71d for details.  
The screenshot you posted uses two’s complement: a value of 1 means Δ=+1, and a value of 127 ≡ -1 (mod 2⁷) means Δ=-1.

**Reply by MickeyGentry** [reply]
I did read it was a relative value but clearly I didn’t understand the implications. Thanks!

**Comment by MickeyGentry** [comment]
Thx for your answer!
